env/work/srw_python/SRWLIB_Example16.py

```diff
@@ -0,0 +1,202 @@
+                           _arUnits=['eV', 'm', 'm', 'ph/s/.1%bw/mm^2'])
+uti_plot.uti_plot2d(arII, plotMeshx, plotMeshy,
+                    ['Horizontal Position [mm]', 'Vertical Position [mm]', 'Intenisty After Propagation [a.u.]'])
+uti_plot.uti_plot1d(arI1y, plotMeshy, ['Vertical Position [mm]', 'Intensity [a.u.]',
+                                       'Intensity After Propagation\n(cut vs vertical position at x=0)'])
+
+srwl.CalcIntFromElecField(arRe, wfr, polarization, 5, 2, wfr.mesh.eStart, 0, 0)
+srwl.CalcIntFromElecField(arIm, wfr, polarization, 6, 2, wfr.mesh.eStart, 0, 0)
+
+parameters = [
+    ['Max Inten B and A propagation', arIinymax, arI1ymax],
+    ['Num of y mesh pts B and A propagation', len(arIinY), len(arI1y)],
+    ['Num of x mesh pts B and A propagation', wfrIn.mesh.nx, wfr.mesh.nx],
+    ['Num of y mesh pts B and A propagation', wfrIn.mesh.ny, wfr.mesh.ny],
+    ['Wfr xS size [mm] B and A propagation', wfrIn.mesh.xStart, wfr.mesh.xStart],
+    ['Wfr xE size [mm] B and A propagation', wfrIn.mesh.xFin, wfr.mesh.xFin],
+    ['Wfr yS size [mm] B and A propagation', wfrIn.mesh.yStart, wfr.mesh.yStart],
+    ['Wfr yE size [mm] B and A propagation', wfrIn.mesh.yFin, wfr.mesh.yFin],
+]
+for i in range(len(parameters)):
+    print('{}{}: [{}, {}]'.format('    ', parameters[i][0], parameters[i][1], parameters[i][2]))
+
+#************************************* 2. Defining parameters for analytic calculation
+lam = 2.4796e-6  # 1.2398/0.5 eV
+numPointsIn = len(arIinY)
+numPointsOut = len(arI1y)
+meshSize = float(wfr.mesh.xFin)
+
+#************************************* 3. Computing intensity distribution as per Born & Wolf, Principles of Optics
+th = []
+sIn = []
+sOut = []
+analyticalIntens = []
+for i in range(numPointsOut):
+    thx = 2.0 * (i - numPointsOut / 2.0 + 0.5) * meshSize / numPointsOut / driftLength
+    th.append(thx)
+    sOut.append(thx * driftLength * 1000)
+try:
+    from scipy.special import jv
+    for i in range(numPointsOut):
+        x = 3.1415 * apertureSize * sin(th[i]) / lam
+        analyticalIntens.append((2 * jv(1, x) / x) ** 2)
+except:
+    pass
+for i in range(numPointsIn):
+    sIn.append(2000.0 * (i - numPointsIn / 2.0) * float(wfrIn.mesh.xFin) / numPointsIn)
+
+#************************************* 4. Plotting
+try:
+    from matplotlib import pyplot as plt
+    fig = plt.figure()
+    ax = fig.add_subplot(111)
+    ax.plot(sIn, arIinY, '--g.', label='SRW (before aperture)')
+    ax.plot(sOut, arI1y, '-r.', label='SRW (after aperture)')
+    if analyticalIntens:
+        ax.plot(sOut, analyticalIntens, '-b.', label='Analytical estimation')
+    ax.legend()
+    ax.set_xlabel('Vertical Position [mm]')
+    ax.set_ylabel('Normalized Intensity, [a.u.]')
+    ax.set_title('Intensity After Propagation\n(cut vs vertical position at x=0)')
+    ax.grid()
+    plt.savefig('{}/compare.png'.format(example_folder))
+    plt.show()
+except:
+    pass
+
+print('done')
```

Comment:
Accepted, after a few cosmetic changes in SRWLIB_Example16.py (use of units in plot2d1d, correcting text of some messages).